plugins/kubernetes/app/models/kubernetes/release_doc.rb

```diff
@@ -104,19 +143,46 @@ def resource_template=(value)
     # dynamically fill out the templates and store the result
     def store_resource_template
       add_pod_disruption_budget
+      add_env_config_map
+
       counter = Hash.new(-1)
+
       self.resource_template = raw_template.map do |resource|
         index = (counter[resource.fetch(:kind)] += 1)
-        TemplateFiller.new(self, resource, index: index).to_hash
+        opts = {index: index}
+        opts[:env_config_map] = env_config_map_name if env_from_config_map?
+        TemplateFiller.new(self, resource, **opts).to_hash
       end
     end
 
+    # The resource we want to copy things from, like name, labels, and annotations
+    def reference_resource
+      return @reference_resource if defined?(@reference_resource)
+      @reference_resource = raw_template.find { |r| %w[Deployment StatefulSet].include?(r[:kind]) }
+    end
```

Comment:
this is usually the `primary` resource and there are already methods in here to find it